docs/reference/ingest/ingest-node.asciidoc

```diff
@@ -577,6 +577,554 @@ value of `service` to the value of the field `code`:
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+If this is the case, use the <<dot-expand-processor, Dot Expand Processor>>
+so that the nested fields may be used in a conditional.
+
+[source,js]
+--------------------------------------------------
+PUT _ingest/pipeline/drop_guests_network
+{
+  "processors": [
+    {
+      "dot_expander": {
+        "field": "network.name"
+      }
+    },
+    {
+      "drop": {
+        "if": "ctx.network?.name == 'Guest'"
+      }
+    }
+  ]
+}
+--------------------------------------------------
+// CONSOLE
+
+Now the following input document can be used with a conditional in the pipeline.
+
+[source,js]
+--------------------------------------------------
+POST test/_doc/3?pipeline=drop_guests_network
+{
+  "network.name": "Guest"
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+////
+Hidden example assertion:
+[source,js]
+--------------------------------------------------
+GET test/_doc/3
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+// TEST[catch:missing]
+
+[source,js]
+--------------------------------------------------
+{
+  "_index": "test",
+  "_type": "_doc",
+  "_id": "3",
+  "found": false
+}
+--------------------------------------------------
+// TESTRESPONSE
+////
+
+The `?.` operators works well for use in the `if` conditional
+because the {painless}/painless-operators-reference.html#null-safe-operator[null safe operator]
+returns null if the object is null and `==` is null safe (as well as many other
+{painless}/painless-operators.html[painless operators]).
+
+However, calling a method such as `.equalsIgnoreCase` is not null safe
+and can result in a NullPointerException.
+
+Some situations allow for the same functionality but done so in a null safe manner.
+For example: `'Guest'.equalsIgnoreCase(ctx.network?.name)` is null safe because
+`Guest` is always non null, but `ctx.network?.name.equalsIgnoreCase('Guest') is not null safe
+since `ctx.network?.name` can return null.
+
+Some situations require an explicit null check. In the following example there
+is not null safe alternative, so an explict null check is needed.
+
+[source,js]
+--------------------------------------------------
+{
+  "drop": {
+    "if": "ctx.network?.name != null && ctx.network.name.contains('Guest')"
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+[[ingest-conditional-complex]]
+=== Complex Conditionals
+The `if` condition can be more then a simple equality check.
+The full power of the <<modules-scripting-painless, Painless Scripting Language>> ability to declare `tag` and `on_failure` fields. These fields are optional
+running in the {painless}/painless-ingest-processor-context.html#null-safe-operator[ingest processor context]
+can be used to determine if the condition should evaluate to `true` or `false`.
+
+IMPORTANT: Values in an `if` condition are read-only and may not change any
```

Comment:
NIT: Wouldn't just: "The value of `ctx` is read-only in `if` conditions. Trying to mutate it will throw and `Exception`." be cleaner? (because it's not "values" that are read-only but just `ctx`)

Reply:
yup, i like your suggestion... updated.